internal/apis/certmanager/types_certificate.go

```diff
@@ -410,8 +410,32 @@ type PKCS12Keystore struct {
 	// PasswordSecretRef is a reference to a key in a Secret resource
 	// containing the password used to encrypt the PKCS12 keystore.
 	PasswordSecretRef cmmeta.SecretKeySelector
+
+	// Algorithms are specifying the key and certificate encryption algorithms and the HMAC algorithm
+	// used to create the PKCS12 keystore. Default value is `LegacyRC2` for backward compatibility.
+	//
+	// If provided, allowed values are:
+	// `LegacyRC2`: Deprecated. Not supported by default in OpenSSL 3 or Java 20.
+	// `LegacyDES`: Less secure algorithm. Use this option for maximal compatibility.
+	// `Modern2023`: Secure algorithm. Use this option in case you have to always use secure algorithms
+	// (eg. because of company policy). Please note that the security of the algorithm is not that important
+	// in reality, because the unencrypted certificate and private key are also stored in the Secret.
+	Algorithms PKCS12Algorithms
```

Comment:
```suggestion
	Algorithm PKCS12Algorithm
```
Would it make sense to rename this field to `Algorithm` instead of `Algorithms`?

Reply:
As it covers a collection of algorithms, i.e. key, certificate encryption algorithm and MAC algorithm. I thought it might be more accurate to use the plural.

Reply:
@SgtCoDFish WDYT?

Reply:
I have no strong preference. I think both work! I have a soft preference for `Algorithm` just because most users won't know or appreciate that there are mulitple algorithms being changed by this so it feels a little more natural to use a singular here although I agree that the plural might be more correct.

We could maybe use something like `Profile` since we're not allowing users to change the underlying algos, but rather to pick one of the profiles we specify. I really don't think it matters much though!

Reply:
@snorwin feel free to unhold if you think the name is ok as-is.

Reply:
I'll rename the field to `Profile`, that should make it ready to be merged.